src/tools/illink/src/ILLink.RoslynAnalyzer/ILLink.RoslynAnalyzer.csproj

```diff
@@ -28,6 +30,7 @@
       <PrivateAssets>all</PrivateAssets>
       <ExcludeAssets>contentfiles</ExcludeAssets> <!-- We include our own copy of the ClosedAttribute to work in source build -->
     </PackageReference>
+    <PackageReference Include="System.Reflection.Metadata" Version="$(SystemReflectionMetadataVersion)" />
```

Comment:
Do we want `SystemReflectionMetadataToolsetVersion` here? Basing off of [this comment](https://github.com/dotnet/runtime/issues/106321#issuecomment-2286785420)